internal/bminventory/inventory.go

```diff
@@ -3760,7 +3757,38 @@ func computeProxyHash(proxy *models.Proxy) (string, error) {
 	return fmt.Sprintf("%x", bs), nil
 }
 
-func validateProxySettings(httpProxy, httpsProxy, noProxy, ocpVersion *string) error {
+func validateProxySettings(params interface{}, ocpVersion *string) error {
+	var (
+		httpProxy       *string
+		httpsProxy      *string
+		noProxy         *string
+		validateVersion bool
+	)
+	switch p := params.(type) {
+	case models.ClusterCreateParams:
+		httpProxy = p.HTTPProxy
+		httpsProxy = p.HTTPSProxy
+		noProxy = p.NoProxy
+		validateVersion = true
+	case models.V2ClusterUpdateParams:
+		httpProxy = p.HTTPProxy
+		httpsProxy = p.HTTPSProxy
+		noProxy = p.NoProxy
+		validateVersion = true
```

Comment:
```suggestion
	case models.ClusterCreateParams, models.V2ClusterUpdateParams:
		httpProxy = p.HTTPProxy
		httpsProxy = p.HTTPSProxy
		noProxy = p.NoProxy
		validateVersion = true
	```

Reply:
it does not work. I have another idea that I'm going to try.

Reply:
https://www.geeksforgeeks.org/golang-program-that-uses-fallthrough-keyword/?ref=rp ?

Reply:
actually not sure that it will work because the type need to be matched to a specific object, it cannot be both